### PR TITLE
fix: updated error message for attempting to deploy a site with a bad…

### PR DESCRIPTION
**Pull request by dylanspyer**
**🎉** Thanks for submitting a pull request! 🎉

#### Summary

Fixes: [6736](https://github.com/netlify/cli/issues/6736)

Previously, `netlify deploy` with a valid site ID and an invalid auth token would log `Error: Site not found. Please rerun "netlify link"`. Updated this error message to be more clear by logging the `message` property from the API response when the response status is 401.

**Before**: 
image

**After**: 
image

---
For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**
image

### CODE REVIEW

**Comment by dylanspyer** [comment]
> Hi @dylanspyer! Thank you so much for your contribution :) could you maybe still log 'site not found' but also render the error message? I think this could inform our users better on what part of the process went wrong and why?

Thanks for the feedback! I pushed an update with a clearer error message:
image

In this failure case, it's possible that the site id is valid and the auth token is invalid. Logging `site not found` in that case might be misleading. Let me know your thoughts :) @khendrikse

File: src/utils/get-site.ts [hunk 1]
```diff
@@ -12,7 +12,13 @@ export const getSiteByName = async (api, siteName) => {
     }
 
     return siteFoundByName
-  } catch {
-    error('Site not found. Please rerun "netlify link"')
+  } catch (error_) {
+    // @ts-expect-error TS(2571) FIXME: Object is of type 'unknown'.
+    if (error_.status === 401) {
+      // @ts-expect-error TS(2571) FIXME: Object is of type 'unknown'.
```
Comment (DanielSLew) [inline src/utils/get-site.ts:18]: Fix these TS errors with pulling in this PR https://github.com/netlify/cli/pull/6877.

**Review by DanielSLew — CHANGES_REQUESTED** [review]
This looks good to me! Just have a request to remove those type errors.

File: src/utils/get-site.ts (hunk 1 above)
Comment (dylanspyer) [reply src/utils/get-site.ts]: Fixed!

**Review by DanielSLew — APPROVED** [review]

**Pull request merged**